qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/parser/template/sections/FragmentSection.java

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+* Copyright (c) 2023 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.parser.template.sections;
+
+import java.util.List;
+
+import com.redhat.qute.parser.template.ASTVisitor;
+import com.redhat.qute.parser.template.Parameter;
+import com.redhat.qute.parser.template.ParameterInfo;
+import com.redhat.qute.parser.template.ParametersInfo;
+import com.redhat.qute.parser.template.Section;
+import com.redhat.qute.parser.template.SectionKind;
+
+/**
+ * Fragment section AST node.
+ * 
+ * <code>
+ 	{#fragment id=item_aliases} 
+	<h2>Aliases</h2>
+	<ol>
+    	{#for alias in aliases}
+    	<li>{alias}</li>
+    	{/for}
+	</ol>
+	{/fragment}
+ * </code>
+ * 
+ * @author Angelo ZERR
+ * 
+ * @see https://quarkus.io/guides/qute-reference#fragments
+ */
+public class FragmentSection extends Section {
+
+	public static final String TAG = "fragment";
+
+	private static final String ID = "id";
+
+	private static final String RENDERED = "rendered";
+
+	private static final ParametersInfo PARAMETER_INFOS = ParametersInfo.builder() //
+			.addParameter(ID) //
+			.addParameter(new ParameterInfo(RENDERED, null, true)) //

```

Comment:
I don't seem to have completion/validation for the parameters (except for `id` being added when I complete the snippet). Is there supposed to be completion/validation? If not, maybe we could look into implementing it (in a different PR).

Reply:
I agree with you, we need this kind of support. Today ParameterInfos are used only to know if a parameter can be assigned to an expression or if the parameter is an expression.

I think it should be nice to use it for completion and validation (with a generic mean). Let's see that in an another PR.